### PR TITLE
Event trace integration

### PR DESCRIPTION
**Pull request by Meorawr**
This adds some hooks to integrate our callback registries into the event trace panel so that we can actually see the events firing through things without needing to revert back to the stone age with print calls and the like. The integration is only enabled in debug mode to avoid any potential conflicts.

image

### CODE REVIEW

File: totalRP3/Core/Debug.lua [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+local function OnCallbackEventFired(registry, event, ...)
+	if not EventTracingRegistries[registry] then
+		return;
+	elseif not EventTrace or not EventTrace.LogLine then
+		return;
+	end
+
+	if TRP3_API.IsEventValid(TRP3_API.GameEvents, event) then
+		return;
+	end
```
Comment (Meorawr) [inline totalRP3/Core/Debug.lua:37]: Will remove this later - this is a leftover from before I added the RemoveFromEventTraceWindow function.

Comment (Solanya) [reply totalRP3/Core/Debug.lua:37]: Are we still removing this later

File: totalRP3/Modules/Map/MapScannersManager.lua [hunk 2]
```diff
@@ -49,7 +41,7 @@ function MapScannersManager.launch(scanID)
 	scan:ResetScanData();
 
 	local function OnScanTimerElapsed()
-		TRP3_Addon:TriggerEvent(Events.MAP_SCAN_ENDED);
+		TRP3_Addon:TriggerEvent("MAP_SCAN_ENDED");
```
Comment (Solanya) [inline totalRP3/Modules/Map/MapScannersManager.lua:44]:
```suggestion
		TRP3_Addon:TriggerEvent(TRP3_Addon.Events.MAP_SCAN_ENDED);
```

File: totalRP3/Modules/Map/MapScannersManager.lua [hunk 3]
```diff
@@ -61,7 +53,7 @@ function MapScannersManager.launch(scanID)
 	end
 
 
-	TRP3_Addon:TriggerEvent(Events.MAP_SCAN_STARTED, scan.duration);
+	TRP3_Addon:TriggerEvent("MAP_SCAN_STARTED", scan.duration);
```
Comment (Solanya) [inline totalRP3/Modules/Map/MapScannersManager.lua:56]:
```suggestion
	TRP3_Addon:TriggerEvent(TRP3_Addon.Events.MAP_SCAN_STARTED, scan.duration);
```

**Review by Solanya — APPROVED** [review]

**Pull request merged**